evm-bridging/src/lib/CrossVMBridgeCallableUpgradeable.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.24;
+
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {ICrossVMBridgeFulfillment} from "../interfaces/ICrossVMBridgeFulfillment.sol";
+
+/**
+ * @title CrossVMBridgeCallable
+ * @dev A base contract intended for use in implementations on Flow, allowing a contract to define
+ *      access to the Cadence X EVM bridge on certain methods.
+ */
+abstract contract CrossVMBridgeCallableUpgradeable is ContextUpgradeable {
```

Comment:
Adapted for upgradable contracts from  [flow-evm-repo draft PR](https://github.com/onflow/flow-evm-bridge/pull/168/files#diff-d5335bccc50b538dd4a4bf2bbadda5300ad0865271ef50aa421411d606a1ca44R12) - the functionality is what's currently expected to register a custom contract in upcoming upgraded bridge